helios-services/src/main/java/com/spotify/helios/master/HostChanges.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.master;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Sets;
+
+import java.util.List;
+import java.util.Set;
+
+class HostChanges  {
+  private final List<String> added;
+  private final List<String> removed;
+
+  HostChanges(final Set<String> newHosts, final Set<String> currentHosts) {
+    this.added = ImmutableList.copyOf(Sets.difference(newHosts, currentHosts).immutableCopy());
```

Comment:
Ah good to know. I felt like I might have been doing a little too much copying here.